Automattic/wp-calypso
# Launchpad: Add client-side handling for calypso_path task data

Related to https://github.com/Automattic/jetpack/pull/32177

## Proposed Changes

* This PR illustrates how we might take advantage of the proposed `calypso_path` task data returned from the server to simplify the client-side logic for tasks.
* The PR _does not_ remove any client-side handling for the time being, as we may have some tasks that aren't up to date, especially on Atomic sites, so it adds some logging via Tracks to identify tasks that don't have a path from the server and we either expect one to be present, or no client-side handler exists.

## Testing Instructions



* Ensure that your back end is running the changes from https://github.com/Automattic/jetpack/pull/32177
* Ensure you can inspect Tracks events
* Navigate to one or more sites that are currently showing a launchpad in Customer Home
* Verify that clicking on tasks that take you to other pages continue to work as expected
* Verify that you see one or more `calypso_launchpad_task_missing_calypso_path` Tracks events, and identify which tasks those are for -- in my testing that includes the newer `write_3_posts` and `enable_subscribers_modal ` tasks.

## Pre-merge Checklist



- [N/A] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [N/A] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [N/A] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
